pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -10849,3 +10849,350 @@ spec:
 		t.Errorf("Expected PipelineRun to be create run failed, but condition reason is %s", reconciledRun.Status.GetCondition(apis.ConditionSucceeded))
 	}
 }
+
+func TestReconciler_PipelineTaskMatrixResultsWithArrayIndexing(t *testing.T) {
+	names.TestingSeed()
+	task := parse.MustParseV1beta1Task(t, `
+metadata:
+  name: mytask
+  namespace: foo
+spec:
+  params:
+    - name: platform
+      default: mac
+  steps:
+    - name: echo
+      image: alpine
+      script: |
+        echo "$(params.platform)"
+`)
+	taskwithresults := parse.MustParseV1beta1Task(t, `
+metadata:
+  name: taskwithresults
+  namespace: foo
+spec:
+  results:
+    - name: platforms
+      type: array
+  steps:
+    - name: produce-a-list-of-platforms
+      image: bash:latest
+      script: |
+        #!/usr/bin/env bash
+        echo -n "[\"linux\",\"mac\",\"windows\"]" | tee $(results.platforms.path)
+`)
+
+	cms := []*corev1.ConfigMap{withEnabledAlphaAPIFields(newFeatureFlagsConfigMap())}
+	cms = append(cms, withMaxMatrixCombinationsCount(newDefaultsConfigMap(), 10))
+	tests := []struct {
+		name                string
+		memberOf            string
+		p                   *v1beta1.Pipeline
+		tr                  *v1beta1.TaskRun
+		expectedTaskRuns    []*v1beta1.TaskRun
+		expectedPipelineRun *v1beta1.PipelineRun
+	}{{
+		name:     "p-dag",
```

Comment:
please give a useful name for this test case
```suggestion
		name:     "indexing results in params",
		name:     "p-dag",
```